File: artefacts/swift-artefact/ios-tuvali-library/Sources/ios-tuvali-library/common/events/EventEmitter.swift
import Foundation

class EventEmitter {
    var listeners: [((Event) -> Void)] = []
    public static let sharedInstance: EventEmitter = EventEmitter()
    
    func emitEvent(_ event: Event) {
        listeners.forEach { listener in
            listener(event)
        }
    }
    
    func emitErrorEvent(message: String, code: String) {
        var eventData: [String: String] = [:]
        eventData["message"] = message
        emitEvent(ErrorEvent(message: message, code: code))
    }
    
    func addListener(listener: @escaping (Event) -> Void) {
        listeners.append(listener)
    }
    
    func removeListeners() {
        listeners.removeAll()
    }
}

File: kotlin/android/src/main/java/io/mosip/tuvali/common/events/EventEmitter.kt
package io.mosip.tuvali.common.events

import io.mosip.tuvali.exception.ErrorCode

class EventEmitter {
  private var listeners = mutableListOf<((Event) -> Unit)>()

  fun addListener(listener: (Event) -> Unit) {
    listeners.add(listener)
  }

  fun emitErrorEvent(message: String, code: ErrorCode) {
    emitEvent(ErrorEvent(message, code.value))
  }

  fun emitEvent(event: Event) {
    listeners.forEach{it(event)}
  }

  fun removeListeners() {
    listeners.clear()
  }
}
